django/db/models/functions/comparison.py

```diff
@@ -159,35 +159,40 @@ def as_sql(self, compiler, connection, **extra_context):
             )
         return super().as_sql(compiler, connection, **extra_context)
 
-    def as_postgresql(self, compiler, connection, **extra_context):
-        copy = self.copy()
-        copy.set_source_expressions(
-            [
-                Cast(expression, TextField()) if index % 2 == 0 else expression
-                for index, expression in enumerate(copy.get_source_expressions())
-            ]
-        )
-        return super(JSONObject, copy).as_sql(
-            compiler,
-            connection,
-            function="JSONB_BUILD_OBJECT",
-            **extra_context,
-        )
-
-    def as_oracle(self, compiler, connection, **extra_context):
+    def as_native(self, compiler, connection, *, returning, **extra_context):
         class ArgJoiner:
             def join(self, args):
-                args = [" VALUE ".join(arg) for arg in zip(args[::2], args[1::2])]
-                return ", ".join(args)
+                pairs = zip(args[::2], args[1::2], strict=True)
```

Comment:
Adding the strict flag here avoids silently dropping the last argument if an odd number of arguments are provided to `JSONObject()`.